src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -26,4 +35,26 @@ public int compare(SongEntity a, SongEntity b) {
       return Double.compare(b.salience, a.salience);
     }
   };
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+
+    if (!(obj instanceof SongEntity)) { 
+      return false; 
+  } 
+
+    SongEntity entity = (SongEntity) obj;
+    return salience == entity.salience
+        && (name == entity.name || (name != null && name.equals(entity.getName())))
```

Comment:
you can just replace it with `Objects.equals(name, entity.name)`. Why you use `entity.name` and then `entity.getName()`?

Reply:
That's how I've seen it done by more people on the internet and I thought:
"Hey, maybe it's a good practice, let's see what the others have to say about it".